File: Stooge  Sort/StoogeSort.java
class StoogeSort {

    static void stoogeSort(int arr[], int low, int high) {

        if(arr[low] > arr[high]) {

            int temp = arr[low];
            arr[low] = arr[high];
            arr[high] = temp;
        }

        if(high - low + 1 > 2) {

            int t = (high - low + 1) / 3;

            stoogeSort(arr, low, high - t);
            stoogeSort(arr, low + t, high);
            stoogeSort(arr, low, high - t);
        }
    }

    public static void main(String[] args) {

        int arr[] = {2, 4, 5, 3, 1};

        stoogeSort(arr, 0, arr.length - 1);

        for(int num : arr)
            System.out.print(num + " ");
    }
}

File: Stooge  Sort/StoogeSort.py
def stooge_sort(arr, low, high):

    if arr[low] > arr[high]:
        arr[low], arr[high] = arr[high], arr[low]

    if high - low + 1 > 2:

        t = (high - low + 1) // 3

        stooge_sort(arr, low, high - t)
        stooge_sort(arr, low + t, high)
        stooge_sort(arr, low, high - t)

arr = [2, 4, 5, 3, 1]

stooge_sort(arr, 0, len(arr) - 1)

print(arr)
